rust/transaction_parsing/src/derivations.rs

```diff
@@ -1,44 +1,129 @@
-    identities::check_derivation_set,
-};
-use definitions::{
-    helpers::unhex, keyring::NetworkSpecsKey, navigation::TransactionCardSet,
-    qr_transfers::ContentDerivations,
+use db_handling::identities::{get_all_addresses, is_passworded, ExportAddrs, ExportAddrsV1};
+
+use definitions::derivations::{
+    DerivedKeyError, DerivedKeyPreview, DerivedKeyStatus, SeedKeysPreview,
 };
+
+use db_handling::helpers::get_network_specs;
+use definitions::helpers::{base58_or_eth_to_multisigner, make_identicon_from_multisigner};
+use definitions::keyring::NetworkSpecsKey;
+use definitions::{helpers::unhex, navigation::TransactionCardSet};
+use parity_scale_codec::Decode;
+use sp_runtime::MultiSigner;
 use std::path::Path;
 
 use crate::cards::Card;
-use crate::error::{Error, Result};
+use crate::error::Result;
 use crate::TransactionAction;
 
 pub fn process_derivations<P>(data_hex: &str, db_path: P) -> Result<TransactionAction>
 where
     P: AsRef<Path>,
 {
     let data = unhex(data_hex)?;
-    let content_derivations = ContentDerivations::from_slice(&data[3..]);
-    let (encryption, genesis_hash, derivations) =
-        content_derivations.encryption_genhash_derivations()?;
-    let network_specs_key = NetworkSpecsKey::from_parts(&genesis_hash, &encryption);
-    let network_specs = try_get_network_specs(&db_path, &network_specs_key)?.ok_or(
-        Error::NetworkForDerivationsImport {
-            genesis_hash,
-            encryption,
-        },
-    )?;
-    check_derivation_set(&derivations)?;
-    let checksum = TrDbColdDerivations::generate(&derivations, &network_specs)
-        .store_and_get_checksum(&db_path)?;
-    let derivations_card = Card::Derivations(&derivations).card(&mut 0, 0);
-    let network_info = network_specs;
+    let export_info = <ExportAddrs>::decode(&mut &data[3..])?;
+    let import_info = prepare_derivations_preview(db_path, export_info)?;
+    let derivations_card = Card::Derivations(&import_info).card(&mut 0, 0);
     Ok(TransactionAction::Derivations {
         content: Box::new(TransactionCardSet {
             importing_derivations: Some(vec![derivations_card]),
             ..Default::default()
         }),
-        network_info: Box::new(network_info),
-        checksum,
-        network_specs_key,
     })
 }
+
+pub fn prepare_derivations_preview<P>(
+    db_path: P,
+    export_info: ExportAddrs,
+) -> Result<Vec<SeedKeysPreview>>
+where
+    P: AsRef<Path>,
+{
+    match export_info {
+        ExportAddrs::V1(v1) => prepare_derivations_v1(db_path, v1),
+    }
+}
+
+fn prepare_derivations_v1<P>(db_path: P, export_info: ExportAddrsV1) -> Result<Vec<SeedKeysPreview>>
+where
+    P: AsRef<Path>,
+{
+    let mut result = Vec::new();
+    for seed_info in export_info.addrs {
+        let mut derived_keys = vec![];
+        for addr_info in seed_info.derived_keys {
+            let multisigner =
+                base58_or_eth_to_multisigner(&addr_info.address, &addr_info.encryption)?;
```

Comment:
And if you have just one eth addr in the pack it will nuke the whole process here.

Reply:
This should not be done at all, the correct way would be something like

1. Arrive at `MultiSigner` value with `seed phrase` + `AddrInfo.derivation_path`;
2. Use `AddrInfo.genesis_hash` + `AddrInfo.encryption` to query the network info and get ss58prefix
3. if needed get the address and compare it to `addr_info.address` for correctness.

tldr: `address` field is only intended for frontends such as Nova or Polkadot-JS to use to do their thing they do where addresses and not keys matter.

Reply:
> 1. Arrive at MultiSigner value with seed phrase + AddrInfo.derivation_path;

This woudn't work for passworded paths.

Reply:
Ok, but it woudn't work for passworded paths.
The only ways to get a `MultiSigner` for a passworded path are:
1. ask user for a password
2. docode `address`

am I missing anything?

Reply:
Ok, lets use `seed phrase + AddrInfo.derivation_path` for passwordless paths.